### PR TITLE
[cgkiele-138,139] Polymorphic difficulty calculation & validation

### PR DESCRIPTION
**Pull request by loverdos**
Some extra features are also present:
* Bump Scala version to `2.12.5`.
* Bump `sbt-verify` version to `0.4.1`.
* Have `sbt-verify` issue warnings instead of errors. This is a temporary solution to some inconsistent behaviour we have observed, so that the developer experience is a bit better.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -54,7 +54,7 @@ val dep = {
 }
 
 val verifyDeps = Seq(
-  "org.scala-lang" % "scala-library" sha1 "dd235d04037dc6f4b6090257872dd35359a563ce",
+  "org.scala-lang" % "scala-library" sha1 "4861b00d921952f2dc9a024198bc72590fd7dc5e",
```
Comment (rtkaczyk) [inline build.sbt]: So, what was the problem/solution?

Comment (loverdos) [reply build.sbt]: `sbt-verify`: locally the last build step ("additional compilation & dist") fails, regardless of Scala version. No solution for that but I think we should decouple the scala version upgrade from whatever `sbt-verify` does. The above changes are just to take into account the modified `2.12.5` scala jars.

Comment (rtkaczyk) [reply build.sbt]: Yes, it fails locally for me too :(

> decouple the scala version upgrade from whatever sbt-verify does

How can we do that? Unless you mean removing `sbt-verify` (or treating the failed checks as warnings) which I would be happy with 😁

File: project/plugins.sbt [hunk 2]
```diff
@@ -6,5 +6,6 @@ addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.0.6")
 addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings" % "latest.release")
 addSbtPlugin("uk.co.josephearl" % "sbt-verify" % "0.3.0-SNAPSHOT")
 addSbtPlugin("com.thesamet" % "sbt-protoc" % "0.99.12")
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.4")
```
Comment (rtkaczyk) [inline project/plugins.sbt]: Interesting, do you intend it for local use only, or should we include it in the build pipeline?

**Comment by rtkaczyk** [comment]
As long as we intend on retaining `sbt-verify` checks could we also use this PR to update its version to `0.4.1`? It's kind of a bug right now...

File: project/plugins.sbt (hunk 2 above)
Comment (loverdos) [reply project/plugins.sbt]: I propose we should (but not necessarily with changes in this PR). It gives you a very good idea of how old your libraries are.

**Comment by loverdos** [comment]
The build is looking good so far, I'll remove the <kbd>`do-not-merge`</kbd> label when it finishes succesfully.

**Pull request merged**